src/Symfony/Component/Mailer/Transport.php

```diff
@@ -93,26 +97,37 @@ public static function fromDsns(#[\SensitiveParameter] array $dsns, ?EventDispat
      */
     public function __construct(
         private iterable $factories,
+        private ?ContainerInterface $rateLimiterLocator = null,
     ) {
     }
 
     public function fromStrings(#[\SensitiveParameter] array $dsns): Transports
     {
         $transports = [];
         foreach ($dsns as $name => $dsn) {
-            $transports[$name] = $this->fromString($dsn);
+            try {
+                $rateLimiter = $this->rateLimiterLocator?->get($name);
```

Comment:
Do we need some sort of a prefix between the name of the transport and the rate limiter?
Like if the name of the transport is `default`, we would look for the `mailer.default` rate limiter?